actionview/test/activerecord/multifetch_cache_test.rb

```diff
@@ -34,4 +34,14 @@ def test_only_preloading_for_records_that_miss_the_cache
     assert_not @topics.detect { |topic| topic.id == topics(:rails).id }.replies.loaded?
     assert     @topics.detect { |topic| topic.id != topics(:rails).id }.replies.loaded?
   end
+
+  def test_preloads_all_records_if_using_cached_proc
+    @view.render partial: "test/partial", collection: [topics(:rails)], cached: proc { |topic| [topic] }
+
+    @topics = Topic.preload(:replies)
```

Comment:
@byroot these tests are flakey, because the preloaded `:replies` sometimes exist across tests. So they only pass if they run in the order of the file, not in the inverse order.

I have a feeling the stuff you did around wrapping the execution context around tests is the key here.

I tried adding `ActiveRecord::Base.connection.clear_query_cache!` in a teardown and that didn't help. I also tried copy pasting https://github.com/rails/rails/blob/b2669d7361fe43e1b1d5ce2add3cb63c9061b761/activesupport/lib/active_support/railtie.rb#L58..L59 and adding it to the test class, also didn't help.

Wondering if you have any tips / thoughts?

Reply:
My backup plan is to refactor this to have an actual controller, make real requests, etc, but that's a bit annoying just for this.